File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/ui/realtime/datasource/DataSourceFactory.java
package com.didichuxing.doraemonkit.ui.realtime.datasource;

/**
 * @desc: 产生数据源对象的工厂类
 */
public class DataSourceFactory {
    public static final int TYPE_NETWORK = 1;
    public static final int TYPE_CPU = 2;
    public static final int TYPE_MEMORY = 3;
    public static final int TYPE_FRAME = 4;

    public static IDataSource createDataSource(int type) {
        switch (type) {
            case TYPE_NETWORK:
                return new NetworkDataSource();
            case TYPE_CPU:
                return new CpuDataSource();
            case TYPE_MEMORY:
                return new MemoryDataSource();
            case TYPE_FRAME:
                return new FrameDataSource();
            default:
                return null;
        }
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/DataSourceFactory.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

/**
 * 产生数据源对象的工厂类
 */
object DataSourceFactory {
    const val TYPE_CPU = 1
    const val TYPE_RAM = 2
    const val TYPE_FPS = 3
    const val TYPE_NETWORK = 4
    fun createDataSource(type: Int): IDataSource {
        return when (type) {
            TYPE_CPU -> CpuDataSource()
            TYPE_RAM -> RamDataSource()
            TYPE_FPS -> FpsDataSource()
            TYPE_NETWORK-> NetworkDataSource()
            else -> DefaultDataSource()
        }
    }
}